### PR TITLE
[SPARK-30962][SQL][DOC]Documentation for Alter table command phase 2

### PR DESCRIPTION
**Pull request by kevinyu98**
### Why are the changes needed?

Based on [JIRA 30962](https://issues.apache.org/jira/browse/SPARK-30962), we want to add all the support `Alter Table` syntax for V1 table.

### Does this PR introduce any user-facing change?

Yes

### How was this patch tested?

Before:
The documentation looks like
 [Alter Table](https://github.com/apache/spark/pull/25590)

After:
Screen Shot 2020-03-03 at 2 02 23 PM
Screen Shot 2020-03-03 at 2 02 41 PM
Screen Shot 2020-03-03 at 2 02 59 PM
Screen Shot 2020-03-03 at 2 03 14 PM
Screen Shot 2020-03-03 at 2 03 28 PM
Screen Shot 2020-03-03 at 2 03 42 PM
Screen Shot 2020-03-03 at 2 03 53 PM

Screen Shot 2020-03-03 at 2 04 03 PM

Notes:
Those syntaxes are not supported by v1 Table.

- `ALTER TABLE .. RENAME COLUMN`
- `ALTER TABLE ... DROP (COLUMN | COLUMNS)`
- `ALTER TABLE ... (ALTER | CHANGE) COLUMN? alterColumnAction` only support change comments, not other actions: `datatype, position, (SET | DROP) NOT NULL`
- `ALTER TABLE .. CHANGE COLUMN?`
- `ALTER TABLE .... REPLACE COLUMNS`
- `ALTER TABLE ... RECOVER PARTITIONS`
-

### CODE REVIEW

**Comment by kevinyu98** [comment]
@huaxingao @dilipbiswal

**Comment by HyukjinKwon** [comment]
ok to test

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 1]
```diff
@@ -83,6 +82,108 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
...
+    <b>Syntax:</b>
+      <code>
+        [ database_name. ] table_name
+      </code>
+  </dd>
+</dl>
+
+<dl>
+  <dt><code><em>COLUMN col_spec</em></code></dt>
+  <dd>Specifies the columns to be altered to be changed.</dd>
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: Nit: ```columns``` -> ```column```?
I guess you missed  ```or``` after ```to be altered```

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 2]
```diff
@@ -83,6 +82,108 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
...
+</dl>
+
+<dl>
+  <dt><code><em>alterColumnAction</em></code></dt>
+   <dd>
+     Change the comment string.<br><br>
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: I saw the following in ```SqlBase.g4```.    Can ```alterColumnAction``` change type, colPosition, setOrDrop as well?
```
alterColumnAction
    : TYPE dataType
    | commentSpec
    | colPosition
    | setOrDrop=(SET | DROP) NOT NULL
    ;
```

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 3]
```diff
@@ -83,6 +82,108 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
...
+     <b>Syntax:</b>
+        <code>
+            COMMENT STRING
+        </code>     
+   </dd>     
+</dl>
+
+
+### ADD AND DROP PARTITION
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: Does ```RECOVER PARTITIONS``` work for V1 as well?

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 4]
```diff
@@ -83,6 +82,108 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
...
+
+#### ADD PARTITION
+`ALTER TABLE ADD` statement adds partition to the partitioned table.
+
+##### Syntax
+{% highlight sql %}
+ALTER TABLE table_identifier ADD [ IF NOT EXISTS ] partition_spec
```
Comment (nchammas) [inline docs/sql-ref-syntax-ddl-alter-table.md]: It's possible to add multiple partitions in one go, so we should show that in the syntax and examples.

For reference, here's how I wrote up the syntax for `ADD PARTITION`: https://github.com/apache/spark/pull/27840/files#diff-1029db085c19beee649492b52c628596R93-R95

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 5]
```diff
@@ -83,6 +82,108 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
 </dl>
 
 
+### ALTER OR CHANGE COLUMN
+`ALTER TABLE ALTER COLUMN` or `ALTER TABLE CHANGE COLUMN` statement changes column's comment.
+
+#### Syntax
+{% highlight sql %}
+ALTER TABLE table_identifier { ALTER | CHANGE } COLUMN col_spec alterColumnAction
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-alter-table.md]: `COLUMN` is optional?

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 6]
```diff
@@ -248,6 +349,86 @@ DESC StudentInfo;
 | age                      | int        | NULL     |
 +--------------------------+------------+----------+--+
 
+-- Add new partition to a table
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-alter-table.md]: nit: `new partition` -> `a new partition`

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 7]
```diff
@@ -248,6 +349,86 @@ DESC StudentInfo;
...
+| partition  |
++------------+--+
+| age=11     |
+| age=12     |
+| age=15     |
++------------+--+
+
+ALTER TABLE StudentInfo ADD IF NOT EXISTS PARTITION (age=18);
+
+-- After adding new partition to the table
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-alter-table.md]: nit: `new partition` -> `a new partition`

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 1 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: Thanks, changed.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 2 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: I tested those: datatype, colPosition, set/drop not null, I got error which states v1 table can't do or only supported with v2 tables. So I didn't put those in the doc.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 3 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: No, it didn't recovery the partition.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 4 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: Yes, it is possible, I will add. Thanks

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 5 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: changed, thanks.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 6 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: done

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 7 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: done

**Review by maropu — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 8]
```diff
@@ -83,6 +82,110 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
...
+### ADD AND DROP PARTITION
+
+#### ADD PARTITION
+`ALTER TABLE ADD` statement adds partition to the partitioned table.
+
+##### Syntax
+{% highlight sql %}
+ALTER TABLE table_identifier ADD [IF NOT EXISTS] 
+  PARTITION <partition_spec> 
+  [ PARTITION <partition_spec> ...]
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: change to 
```
ALTER TABLE table_identifier ADD [ IF NOT EXISTS ] 
  ( partition_spec [ , partition_spec ... ] )
```
so the format is consistent with others?

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 9]
```diff
@@ -274,5 +482,6 @@ ALTER TABLE dbx.tab1 UNSET TBLPROPERTIES ('winner')
 ### Related Statements
 - [CREATE TABLE](sql-ref-syntax-ddl-create-table.html)
 - [DROP TABLE](sql-ref-syntax-ddl-drop-table.html)
+- [ALTER TABLE ... ADD PARTITION](sql-ref-syntax-ddl-alter-table.html)
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: is this change intentional?

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 8 above)
Comment (nchammas) [reply docs/sql-ref-syntax-ddl-alter-table.md]: `,` is not part of the syntax in Spark.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 9 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: oh, I thought that one is part of umbrella jira, so tried to reference. Thanks for catching this, I will remove it.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 8 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: I will make changes. Thanks

**Review by srowen — COMMENTED** [review]
Text looks OK to me.

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 8 above)
Comment (huaxingao) [reply docs/sql-ref-syntax-ddl-alter-table.md]: @nchammas You are right. No ```,```. Thanks for catching this.

File: docs/sql-ref-syntax-ddl-alter-table.md [hunk 10]
```diff
@@ -83,6 +82,109 @@ ALTER TABLE table_identifier ADD COLUMNS ( col_spec [ , col_spec ... ] )
 </dl>
 
 
+### ALTER OR CHANGE COLUMN
+`ALTER TABLE ALTER COLUMN` or `ALTER TABLE CHANGE COLUMN` statement changes column's comment.
+
+#### Syntax
+{% highlight sql %}
+ALTER TABLE table_identifier { ALTER | CHANGE } [COLUMN] col_spec alterColumnAction
```
Comment (huaxingao) [inline docs/sql-ref-syntax-ddl-alter-table.md]: super nit: ```[COLUMN]``` ->``` [ COLUMN ]```

Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: done

Comment (maropu) [reply docs/sql-ref-syntax-ddl-alter-table.md]: You forgot to address the @huajianmao comment?

**Comment by maropu** [comment]
Pending the fix: https://github.com/apache/spark/pull/27779#discussion_r390490103

File: docs/sql-ref-syntax-ddl-alter-table.md (hunk 10 above)
Comment (kevinyu98) [reply docs/sql-ref-syntax-ddl-alter-table.md]: done

**Comment by maropu** [comment]
Thanks! Merged to master/branch-3.0. cc: @gatorsmile
